src/NCronJob/Scheduler/JobWorker.cs

```diff
@@ -68,7 +68,7 @@ await DispatchJobForProcessing(nextJob, priorityTuple.NextRunTime, queueName, se
     public async Task InvokeJobWithSchedule(JobRun jobRun, CancellationToken cancellationToken)
     {
         jobRun.NotifyStateChange(JobStateType.Scheduled);
-        await WaitForNextExecution(jobRun.RunAt ?? DateTimeOffset.UtcNow, cancellationToken).ConfigureAwait(false);
```

Comment:
Yeah. I really didn't make sense to have `RunAt` as nullable. Maybe we should also call it `StartedAt` rather than `RunAt` because this leaves much room for interpretation.

Reply:
Because of the InstantRegistry use case, I believe we should still keep it named RunAt, at it may hold a date in the future.